x/dex/keeper/grpc_query_simulate_cancel_limit_order.go

```diff
@@ -0,0 +1,40 @@
+package keeper
+
+import (
+	"context"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/neutron-org/neutron/v4/x/dex/types"
+)
+
+func (k Keeper) SimulateCancelLimitOrder(
+	goCtx context.Context,
+	req *types.QuerySimulateCancelLimitOrderRequest,
+) (*types.QuerySimulateCancelLimitOrderResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+	cacheCtx, _ := ctx.CacheContext()
```

Comment:
An interesting question: does this uncommited cached context consume gas during execution in contracts?

In other words, is this simulation free for contracts during their execution?

Reply:
My understanding is that cacheCtx still consumes gas. So, yes, contract will pay the query gas cost